### ISSUE TITLE
Add decomposed Ridge-load + HGBR-PV forecaster

### ISSUE THREAD

**Issue by jesuscontrerasocana** [jesuscontrerasocana/scaffold#32]
## Goal

Add a new selectable decomposed forecaster that keeps the current Ridge load model unchanged and replaces only the PV component with a **single global `HistGradientBoostingRegressor`** shared across all forecast leads.

The existing `ridge_decomposed` model should remain available and unchanged so the new approach can be evaluated cleanly against the current baseline.

The initial implementation used one HGBR per forecast lead (132 models). In simulation this was far too slow: roughly 5 seconds per simulated hour, driven by 132 separate sklearn `predict()` calls at every decision. Replace that design with one global PV model that predicts all 132 lead rows in one vectorized call.

## New model

Add/keep a separate model option, e.g.:

```text
ridge_hgbr_decomposed
```

with:

- **Load forecast:** existing decomposed Ridge load model
- **PV forecast:** one global `HistGradientBoostingRegressor` across all leads
- **Net-load forecast:** `load_forecast - pv_forecast`
- same 132-step / 33-hour forecast interface
- train once, persist, reload, then infer during simulation
- no optimizer changes

Do not replace or alter `ridge_decomposed`.

## PV training data

Train the HGBR on rows representing `(decision_time, forecast_lead)` pairs rather than fitting one separate model per lead.

For each training decision and each valid lead, construct one row with the target PV at that lead.

The model should therefore learn across all horizons and share structure between leads.

## PV features

Use the existing PV information plus an explicit forecast-horizon feature:

- `most_recent_load_factor_forecast` at the target timestamp
- `time_of_day_sin`
- `time_of_day_cos`
- PV lag 15 min at decision time
- PV lag 1 h at decision time
- `lead_steps` (1–132), or equivalently lead time in hours

Keep the first implementation deliberately small. Only add extra seasonality such as month or annual sin/cos if the initial experiment shows a clear need.

Preserve leakage-safe feature construction: only observations strictly before `at_time` may be used as historical inputs.

## HGBR structure

Use **one** `HistGradientBoostingRegressor`, not 132 separate models.

Prefer a straightforward implementation over a large hyperparameter search. Use a small, documented parameter set and keep training/inference simple enough for the monthly simulation runtime constraint.

At inference, build the full 132-row PV feature frame and make a single model call:

```python
pv_prediction = pv_model.predict(pv_features)
```

Do not loop over leads and call sklearn `predict()` once per lead.

The trained PV model must be persisted and restored by `save` / `load`.

Clip PV predictions at zero before recombining with load, consistent with the current decomposed forecaster.

Avoid training Ridge PV models that are immediately discarded. Share/refactor only the small amount of common load-training / feature setup needed by both decomposed variants rather than calling a parent fit path that trains the full Ridge PV stack unnecessarily.

## Model selection

Keep `ridge_decomposed` as the default selected model until the HGBR variant has been benchmarked and shown to be preferable.

The HGBR model should be explicitly selectable for experiments.

## Evaluation

Compare the new model against `ridge_decomposed` on the same development period and report at least:

### Net load
- MAE
- RMSE
- bias
- normalized MAE

### Components
- load MAE
- PV MAE
- PV RMSE if useful

Also inspect performance by forecast lead, especially:

- lead 1 / 15 min
- first hour
- first 4 hours
- full 132-step horizon

Short-horizon PV accuracy matters disproportionately because only the first battery action is executed before re-optimization.

Current decomposed Ridge reference is approximately:

```text
Net MAE:   12.11 kW
Net RMSE:  18.87 kW
Net bias:   0.14 kW
Net nMAE:   0.460

Load MAE:   4.22 kW
PV MAE:    10.86 kW
```

## Runtime

Record forecast runtime for the June simulation and compare against the current decomposed Ridge model.

At minimum report:

```text
n_decisions
forecast_seconds_total
forecast_seconds_per_decision
wall_clock_seconds
```

The previous 132-HGBR implementation was not viable (~5 seconds per simulated hour). The global-model version should remove that per-lead sklearn call overhead and remain comfortably within the assignment runtime budget.

## Tests

Add tests covering:

- full 132-step output with no NaNs;
- output index contract;
- exactly one HGBR model is used for PV;
- inference performs one HGBR prediction call for the full horizon rather than one call per lead;
- no leakage from history at or after `at_time`;
- `lead_steps` / horizon feature is constructed correctly;
- PV predictions clipped at zero;
- save/load forecast equivalence;
- existing `ridge_decomposed` behavior remains unchanged;
- model selection can explicitly choose both `ridge_decomposed` and the new HGBR variant.

## Out of scope

Do not include short-horizon PV bias/error correction in this issue. That should remain a separate follow-up experiment if HGBR still leaves meaningful short-horizon error.

Do not change:

- optimizer behavior;
- load Ridge feature definitions or training semantics;
- forecast horizon;
- existing `ridge_decomposed` model semantics.

## Acceptance criteria

- New selectable Ridge-load + **single global HGBR-PV** decomposed forecaster exists.
- Existing `ridge_decomposed` remains unchanged, selectable, and the default until comparison is complete.
- One global HGBR is trained across all forecast leads using a lead/horizon feature.
- One vectorized HGBR `predict()` call produces the PV horizon at each decision.
- No unnecessary Ridge PV stack is trained for the HGBR variant.
- Model trains once, persists, reloads, and simulates correctly.
- Leakage-safe PV features are used.
- PV predictions are non-negative before recombination.
- Forecast-quality comparison against `ridge_decomposed` is reported.
- June runtime is reported and is practical for the assignment.
- Existing tests pass.
